test/e2e/operator/operator.go

```diff
@@ -16,40 +16,116 @@
 package inteldevicepluginsoperator
 
 import (
+	"path/filepath"
+	"time"
+
 	"github.com/intel/intel-device-plugins-for-kubernetes/test/e2e/utils"
 	"github.com/onsi/ginkgo/v2"
-	"github.com/onsi/gomega"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
 	admissionapi "k8s.io/pod-security-admission/api"
 )
 
 const (
-	kustomizationYaml = "deployments/operator/default/kustomization.yaml"
+	ns      = "inteldeviceplugins-system"
+	timeout = time.Second * 300
 )
 
 func init() {
 	ginkgo.Describe("Device Plugins Operator", describe)
 }
 
+func kubectl(ns string, cmd string, opt string, file string) {
+	path, err := utils.LocateRepoFile(file)
+	if err != nil {
+		framework.Failf("unable to locate %q: %v", file, err)
+	}
+
+	if opt == "-k" {
+		path = filepath.Dir(path)
+	}
+
+	msg := framework.RunKubectlOrDie(ns, cmd, opt, path)
+	framework.Logf("Delete node-feature-discovery:\n%s", msg)
+}
+
 func describe() {
-	f := framework.NewDefaultFramework("inteldevicepluginsoperator")
+	f := framework.NewDefaultFramework(ns)
 	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
 
-	var webhook v1.Pod
-
 	ginkgo.BeforeEach(func() {
-		kustomizationPath, err := utils.LocateRepoFile(kustomizationYaml)
-		if err != nil {
-			framework.Failf("unable to locate %q: %v", kustomizationYaml, err)
+		kubectl("node-feature-discovery", "apply", "-k", "deployments/nfd/kustomization.yaml")
+
+		kubectl("node-feature-discovery", "apply", "-k", "deployments/nfd/overlays/node-feature-rules/kustomization.yaml")
+
+		kubectl(ns, "apply", "-k", "deployments/operator/default/kustomization.yaml")
+
+		waitForPod(f, "controller-manager")
+
+		time.Sleep(timeout)
+	})
+
+	ginkgo.AfterEach(func() {
+		kubectl(ns, "delete", "-k", "deployments/operator/default/kustomization.yaml")
+
+		kubectl("node-feature-discovery", "delete", "-k", "deployments/nfd/kustomization.yaml")
+	})
+
+	ginkgo.It("deploys DSA plugin", func() {
```

Comment:
FWIW, I'd be OK to replace the existing plugin deployments that use kustomize with the operator based one. After all, the kustomization .yaml's are built into the operator so they are implicitly tested.

Reply:
This would also be closer to what we agreed earlier: have these operator tests in their own device specific containers (ginkgo term) instead of in a single operator one.